hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/BlockDataStreamOutput.java

```diff
@@ -259,6 +268,19 @@ public void write(ByteBuffer b, int off, int len) throws IOException {
     }
 
     final StreamBuffer buf = new StreamBuffer(b, off, len);
+
+    // Controls the number of buffers on the client.
+    if (futures.size() >= streamWindowSize) {
+      try {
+        futures.poll().get();
```

Comment:
Q. Does this mean that once this window is reached i.e once 64 buffers get filled , it will keep blocking (waiting for the future to complete) the client for every buffer till the last one is written?

Reply:
Yeah, if it's full, let's wait for the first one to finish, and then we'll put another buf to the bufferList